src/components/DownloadImageDialog/ImageForm.tsx

```diff
@@ -204,347 +251,550 @@ export const ImageForm = memo(function ImageForm({
+						<>
+							<InputLabel htmlFor="device-wifi-ssid" sx={{ mb: 2 }}>
+								WiFi SSID
+							</InputLabel>
+							<TextField
+								value={model.wifiSsid}
+								id="device-wifi-ssid"
+								slotProps={{
+									htmlInput: {
+										name: 'wifiSsid',
+										autoComplete: 'wifiSsid-auto-complete',
+									},
+								}}
+								onChange={(event) => {
+									onChange('wifiSsid', event.target.value);
+								}}
+							/>
+							<InputLabel htmlFor="device-wifi-password" sx={{ my: 2 }}>
+								Wifi Passphrase
+							</InputLabel>
+							<TextField
+								type={showPassword ? 'text' : 'password'}
+								id="device-wifi-password"
+								value={model.wifiKey}
+								slotProps={{
+									htmlInput: {
+										name: 'wifiKey',
+									},
+									// input and htmlInput are different https://mui.com/material-ui/api/text-field/#text-field-prop-slotProps
+									input: {
 										endAdornment: (
-											<>
-												{version.value === recommendedVersion && (
-													<Chip
-														sx={{ ml: 1 }}
-														color="green"
-														label="recommended"
-													/>
-												)}
-												{!!version?.knownIssueList && (
-													<Tooltip title={version.knownIssueList}>
-														<FontAwesomeIcon
-															icon={faTriangleExclamation}
-															color={token('color.icon.warning')}
-														/>
-													</Tooltip>
-												)}
-												{InputProps.endAdornment}
-											</>
+											<InputAdornment position="end">
+												<IconButton
+													onClick={() => {
+														setShowPassword((show) => !show);
+													}}
+													onMouseDown={(
+														event: React.MouseEvent<HTMLButtonElement>,
+													) => {
+														event.preventDefault();
+													}}
+													edge="end"
+												>
+													{showPassword ? <VisibilityOff /> : <Visibility />}
+												</IconButton>
+											</InputAdornment>
 										),
-									}}
-								/>
-							)}
-							disableClearable
+									},
+								}}
+								onChange={(event) => {
+									onChange('wifiKey', event.target.value);
+								}}
+							/>
+						</>
+					)}
+				</Stack>
+				{supportsSecureBoot && (
+					<>
+						<Divider
+							variant="fullWidth"
+							sx={{ my: 3, borderStyle: 'dashed' }}
 						/>
-					</Box>
-					{showAllVersionsToggle && (
-						<Box
-							mx={2}
-							display="flex"
-							alignItems="center"
-							alignSelf="flex-end"
-							// TODO: find a better way to center the checkbox with the input only (without label)
-							height={54}
-						>
+						<FormControl>
+							<FormLabel id="secure-boot-and-full-disk-encryption-label">
+								<Stack direction="row" alignItems="center" gap={1}>
+									<Typography variant="titleSm">
+										Secure Boot and Full Disk Encryption
+									</Typography>
+									{/* TODO: Pick an actualy expiry date */}
```

Comment:
can we solve this TODO or we need some extra info ?

Reply:
We still don't know when we're shipping this. Once we pick a date to ship, I'll set an expiry date at least a month later